### PR TITLE
QA-487: test: Split out cross-platform tests

### PR DESCRIPTION
**Pull request by lluiscampos**
Update image-tests submodule to bring the split functionality.
    
    Mark all tests accordingly. Along the way, minor tests doc improvements
    have been made.

### CODE REVIEW

**Comment by lluiscampos** [comment]
Ready for review!

These three go together:
* https://github.com/mendersoftware/mender-image-tests/pull/80
* https://github.com/mendersoftware/meta-mender/pull/1884
* https://github.com/mendersoftware/mender-qa/pull/656

As agreed with the client team, I wanted to gather some runtime duration numbers to see how much of an improvement or not this is. However, while working on it we also have changed the disk types that our workers use for client acceptance tests, so I had to re-launch builds.

The table below summarizes with and without SSD the running times. For the case with SSD I only have one sample, though.

Builds:
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/750461351 (no SSD)
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/750461495 (no SSD)
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/751610594 (yes SSD)

Nightly builds used to compare against:
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/750181769 (no SSD)
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/748936973 (no SSD)
* https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/751380551 (yes SSD)

The numbers :nerd_face: 

See the spreadsheet [here](https://docs.google.com/spreadsheets/d/102Iyu-5zppHKXrMUYbmXLJ0CYh9bKL2qDbmt4OydSNs/edit#gid=1147316915).

image

The new job, `test:acceptance:qemux86_64:uefi_grub:cross-platform` will be the slowest. Without SSD actually it didn't make much sense (it took the same time as the old `test:acceptance:qemux86_64:uefi_grub`) but with SSD it does shave off like 30+ minutes (although adding a +30 minutes in total CI time).

So... what do you think?

File: tests/acceptance/test_build.py [hunk 1]
```diff
@@ -112,6 +114,7 @@ def test_certificate_split(self, request, bitbake_image, prepared_test_build):
 
         assert md5sums == output
 
+    @pytest.mark.cross_platform
     @pytest.mark.only_with_image("sdimg")
```
Comment (kacf) [inline tests/acceptance/test_build.py:118]: This one must be kept, `cross_platform` is incompatible with `sdimg`.

File: tests/acceptance/test_dataimg.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 from utils.common import build_image, latest_build_artifact, make_tempdir
 
 
+@pytest.mark.cross_platform
```
Comment (kacf) [inline tests/acceptance/test_dataimg.py:24]: Keep this one, it's also testing ubi (it has no `only_with_image` at all).

File: tests/acceptance/test_inventory.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 
 @pytest.mark.usefixtures("setup_board", "bitbake_path")
 class TestInventory:
+    @pytest.mark.cross_platform
```
Comment (kacf) [inline tests/acceptance/test_inventory.py:26]: Keep this one, it's testing that our inventory script is returning the correct filesystem type, very relevant for UBI.

**Review by kacf — APPROVED** [review]
Nice work!

All comments can be easily fixed, and do not require re-review, so pre-approving.

**Comment by lluiscampos** [comment]
@mender-test-bot cherry-pick to:
* kirkstone

**Pull request merged**